homeassistant/components/zwave_js/binary_sensor.py

```diff
@@ -0,0 +1,351 @@
+        )
+    )
+
+
+class ZWaveBooleanBinarySensor(ZWaveBaseEntity, BinarySensorEntity):
+    """Representation of a Z-Wave binary_sensor."""
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the sensor is on or off."""
+        return bool(self.info.primary_value.value)
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return device class."""
+        if self.info.primary_value.command_class == CommandClass.BATTERY:
+            return DEVICE_CLASS_BATTERY
+        return None
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        if self.info.primary_value.command_class == CommandClass.SENSOR_BINARY:
+            # Legacy binary sensors are phased out (replaced by notification sensors)
+            # Disable by default to not confuse users
+            if self.info.node.device_class.generic != "Binary Sensor":
+                return False
+        return True
+
+
+class ZWaveNotificationBinarySensor(ZWaveBaseEntity, BinarySensorEntity):
+    """Representation of a Z-Wave binary_sensor from Notification CommandClass."""
+
+    def __init__(self, client: ZwaveClient, info: ZwaveDiscoveryInfo) -> None:
+        """Initialize a ZWaveNotificationBinarySensor entity."""
+        super().__init__(client, info)
+        # check if we have a custom mapping for this value
+        self._mapping_info = self._get_sensor_mapping()
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the sensor is on or off."""
+        if self._mapping_info:
+            return self.info.primary_value.value in self._mapping_info["states"]
+        return bool(self.info.primary_value.value != 0)
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return device class."""
+        return self._mapping_info.get("device_class")
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        # We hide some more advanced sensors by default to not overwhelm users
+        return self._mapping_info.get("enabled", True)
+
+    @callback
+    def _get_sensor_mapping(self) -> NotificationSensorMapping:
+        """Try to get a device specific mapping for this sensor."""
+        for mapping in NOTIFICATION_SENSOR_MAPPINGS:
+            if mapping["type"] != int(
+                self.info.primary_value.metadata.cc_specific["notificationType"]
+            ):
+                continue
+            for state_key in self.info.primary_value.metadata.states:
+                # make sure the key is int
+                state_key = int(state_key)
+                if state_key not in mapping["states"]:
+                    continue
+                # match found
+                mapping_info = mapping.copy()
+                return mapping_info
+        return {}
```

Comment:
Is this case just when we haven't added it yet to the mapping?

Reply:
yeah, it's impossible to predict all possible mappings. I did my best to map the most common/used.
btw: I'll add a quick adjustment to disable sensors for which we do not have a mapping as those are considered to be advanced.

Reply:
checked in that fixed